src/Language/PureScript/Sugar/TypeClasses/Deriving.hs

```diff
@@ -733,11 +737,22 @@ mkVarMn mn = Var . Qualified mn
 mkVar :: Ident -> Expr
 mkVar = mkVarMn Nothing
 
+-- This function may seem a little obtuse, but it's only this way to ensure
+-- that it is injective. Injectivity is important here; without it, we can end
+-- up with accidental variable shadowing in the generated code.
```

Comment:
Perhaps I've made an incorrect assumption about how `Ident` generation works, actually; I'm not completely sure whether this comment is correct. If it really worked how I am imagining then I think `lamNull` above should break for the same reason that the previous code did?